### PR TITLE
Misha17.1var2 endpoint for delete user from workspace

### PR DESCRIPTION
**Pull request by DoctorHeisenberg**

### CODE REVIEW

File: Provis.Core/Services/WorkspaceService.cs [hunk 1]
```diff
@@ -265,5 +266,41 @@ public async Task<WorkspaceInfoDTO> GetWorkspaceInfoAsync(int workspId, string u
...
+            if (user == null)
+            {
+                throw new HttpException(System.Net.HttpStatusCode.NotFound, 
+                    "User with this Id not exist");
+            }
+
+            var workspace = _workspaceRepository
+                .Query()
+                .Where(x => x.Id == workspaceId)
+                .FirstOrDefault();
```
Comment (pashunskyi) [inline Provis.Core/Services/WorkspaceService.cs]: maby better do "_workspaceRepository.GetByKeyAsync(workspaceId)"?

Comment (DoctorHeisenberg) [reply Provis.Core/Services/WorkspaceService.cs]: you god damn right

**Comment by YevheniiVovk** [comment]
@DoctorHeisenberg  what about logic related to assign task to user ?

File: Provis.Core/Services/WorkspaceService.cs [hunk 2]
```diff
@@ -265,5 +266,38 @@ public async Task<WorkspaceInfoDTO> GetWorkspaceInfoAsync(int workspId, string u
...
+                    "User with this Id not exist");
+            }
+
+            var workspace = _workspaceRepository.GetByKeyAsync(workspaceId);
+
+            if (workspace == null)
+            {
+                throw new HttpException(System.Net.HttpStatusCode.NotFound, 
+                    "Workspace with Id not exist");
+            }
```
Comment (YevheniiVovk) [inline Provis.Core/Services/WorkspaceService.cs]: Let's remove not needed request to DB

File: Provis.Core/Services/WorkspaceService.cs [hunk 3]
```diff
@@ -265,5 +266,38 @@ public async Task<WorkspaceInfoDTO> GetWorkspaceInfoAsync(int workspId, string u
...
+
+            var userWorksp = _userWorkspaceRepository
+                .Query()
+                .Where(x => x.WorkspaceId == workspaceId && x.User.Email == deleteUserDTO.UserEmail)
```
Comment (YevheniiVovk) [inline Provis.Core/Services/WorkspaceService.cs]: Please optimaze the request to db, look on execution plan and make sure that indexes seek is used for it

File: Provis.WebApi/Controllers/WorkspaceController.cs [hunk 4]
```diff
@@ -92,5 +92,17 @@ public async Task<IActionResult> GetWorkspaceInfoAsync(int id)
...
         }
+
+        [Authorize]
+        [HttpDelete]
+        [WorkspaceRoles(new WorkSpaceRoles[] { WorkSpaceRoles.OwnerId })]
+        [Route("{workspaceId}/user")]
+        public async Task<IActionResult> DeleteFromWorkspace(int workspaceId, 
+            [FromBody] DeleteUserDTO deleteUserDTO)
+        {
+            await _workspaceService.DeleteFromWorkspaceAsync(workspaceId, deleteUserDTO);
```
Comment (YevheniiVovk) [inline Provis.WebApi/Controllers/WorkspaceController.cs:104]: Add validator

**Review by vladtymo — APPROVED** [review]

**Review by YevheniiVovk — APPROVED** [review]

**Pull request closed without merging**